.github/workflows/ci.yml

```
@@ -8,20 +8,26 @@ on:
 
 jobs:
  checks:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+
     strategy:
       fail-fast: false
       matrix:
-        python-version:
-        - 2.7
-        - 3.5
-        - 3.6
-        - 3.8
-        - 3.9
-    name: Check Python ${{ matrix.python-version }}
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        python-version: ['2.7', '3.5', '3.6', '3.7', '3.8', '3.9']
+        exclude:
+          - os: macos-latest
+            python-version: '3.7'
+          - os: macos-latest
+            python-version: '3.8'
+          - os: windows-latest
+            python-version: '3.7'
+          - os: windows-latest
+            python-version: '3.8'
```

This is a very large matrix, and is about to get bigger if we add Python 3.10 (now if we kill python 2 and 3.5... ;) ) MacOS builds are limited to 5x at a time, and limits are shared across an org, I believe. How about just adding a few, rather than excluding a few? Also, no one uses Python 2.7 on Windows, most binaries packages for it are frozen due to Microsoft pulling the 2008 compiler download. For pure-Python things like this, it's pretty unlikely to have major differences.

```suggestion
        os: [ubuntu-latest]
        python-version: ['2.7', '3.5', '3.6', '3.7', '3.8', '3.9']
        include:
          - {os: macos-latest, python-version: '2.7'}
          - {os: macos-latest, python-version: '3.9'}
          - {os: windows-latest, python-version: '3.7'}
          - {os: windows-latest, python-version: '3.9'}
        exclude:
          - {os: ubuntu-latest, python-version: '3.7'}
```

---

Right, one would never continue adding without the removal of some. That's only fair. But agreed that the matrix can be made fairly smaller.